egs/librispeech/asr/simple_v1/bpe_ctc_att_conformer_decode.py

```diff
@@ -6,83 +6,301 @@
 import argparse
 import logging
 import os
-import random
-import re
-import sys
 
+from collections import defaultdict
 from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
 from typing import Union
 
 import k2
 import numpy as np
 import torch
 
-from snowfall.data import LibriSpeechAsrDataModule
+from k2 import Fsa, SymbolTable
+
 from snowfall.common import average_checkpoint, store_transcripts
+from snowfall.common import find_first_disambig_symbol
 from snowfall.common import get_texts
 from snowfall.common import load_checkpoint
 from snowfall.common import str2bool
 from snowfall.common import write_error_stats
+from snowfall.data import LibriSpeechAsrDataModule
+from snowfall.decoding.graph import compile_HLG
+from snowfall.decoding.lm_rescore import rescore_with_n_best_list
+from snowfall.decoding.lm_rescore import rescore_with_whole_lattice
+from snowfall.models import AcousticModel
 from snowfall.models.conformer import Conformer
 from snowfall.text.numericalizer import Numericalizer
 from snowfall.training.ctc_graph import build_ctc_topo
+from snowfall.training.mmi_graph import get_phone_symbols
+
+
+def nbest_decoding(lats: k2.Fsa, num_paths: int):
```

Comment:
I assume this nbest_decoding function is still here as some kind of demo?  It didn't help vs. just one-best, right?

Reply:
I plan to combine this nbest_decoding with transformer-decoder nbest-rescore. 
Now only encoder model is used, and transformer-decoder model may be used as a rescore "Language model".
I am still working on this.